NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -19,40 +24,544 @@
+    ) * units.rad
+
+    # in CORSIKA convention, the first component points North (y in NRR) and the second component points down (minus z)
+    By, minBz = corsika['inputs'].attrs["MAGNET"]
+    B_inclination = np.arctan2(minBz, By)  # angle from y-axis towards negative z-axis
+
+    B_strength = np.sqrt(By ** 2 + minBz ** 2) * units.micro * units.tesla
+
+    # zenith of the magnetic field vector is 90 deg + inclination, as inclination proceeds downwards from horizontal
+    # azimuth of the magnetic field vector is 90 deg - declination, as declination proceeds clockwise from North
+    magnetic_field_vector = B_strength * hp.spherical_to_cartesian(
+        np.pi / 2 + B_inclination, np.pi / 2 - declination / units.rad
+    )
 
     return zenith, azimuth, magnetic_field_vector
 
 
+def get_geomagnetic_angle(zenith, azimuth, magnetic_field_vector):
+    """
+    Calculates the angle between the geomagnetic field and the shower axis defined by `zenith` and `azimuth`.
+
+    Parameters
+    ----------
+    zenith : float
+        zenith angle (in internal units)
+    azimuth : float
+        azimuth angle (in internal units)
+    magnetic_field_vector : np.ndarray
+        The magnetic field vector in the NRR coordinate system (x points East, y points North, z points up)
+
+    Returns
+    -------
+    geomagnetic_angle : float
+        geomagnetic angle
+    """
+    shower_axis_vector = hp.spherical_to_cartesian(zenith / units.rad, azimuth / units.rad)
+    geomagnetic_angle = hp.get_angle(magnetic_field_vector, shower_axis_vector) * units.rad
+
+    return geomagnetic_angle
+
+
+def convert_obs_to_nuradio_efield(observer, zenith, azimuth, magnetic_field_vector):
+    """
+    Converts the electric field from one CoREAS observer to NuRadio units and the on-sky coordinate system.
+
+    The on-sky CS in NRR has basis vectors eR, eTheta, ePhi.
+    Before going to the on-sky CS, we account for the magnetic field which does not point strictly North.
+    To get the zenith, azimuth and magnetic field vector, one can use `get_angles()`.
+    The `observer` array should have the shape (n_samples, 4) with the columns (time, Ey, -Ex, Ez),
+    where (x, y, z) is the NuRadio CS.
+
+    Parameters
+    ----------
+    observer : np.ndarray
+        The observer as in the HDF5 file, e.g. list(corsika['CoREAS']['observers'].values())[i].
+    zenith : float
+        zenith angle (in internal units)
+    azimuth : float
+        azimuth angle (in internal units)
+    magnetic_field_vector : np.ndarray
+        magnetic field vector
+    
+    Returns
+    -------
+    efield: np.array (3, n_samples)
+        Electric field in the on-sky CS (r, theta, phi)
+    efield_times: np.array (n_samples)
+        The time values corresponding to the electric field samples
+
+    """
+    cs = coordinatesystems.cstrafo(
+        zenith / units.rad, azimuth / units.rad,
```

Comment:
also here, unit multiplication eeds to be removed.

Reply:
I had this discussion with Mitja once. He argued that this can not break things, it would im principle allow us to change default units (although we never would do this because all hell would break loose), and it is very explicit. Are we overlooking anything?